File: lib/features/multiplayer/services/room_code_service.dart
import 'dart:math';

class RoomCodeService {
  RoomCodeService({Random? random}) : _random = random ?? Random();

  static const _characters = 'ABCDEFGHJKLMNPQRSTUVWXYZ23456789';
  final Random _random;

  String generateRoomCode() => List.generate(
    6,
    (_) => _characters[_random.nextInt(_characters.length)],
  ).join();

  bool isValidRoomCode(String code) =>
      RegExp(r'^[A-Z0-9]{6}$').hasMatch(code.trim().toUpperCase());
}

File: backend/src/services/room_code.service.js
const crypto = require("crypto");

const CHARACTERS = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789";

function generateRoomCode() {
  const bytes = crypto.randomBytes(6);
  return Array.from(bytes, (byte) => CHARACTERS[byte % CHARACTERS.length]).join(
    "",
  );
}

module.exports = { generateRoomCode };
